woocommerce/woocommerce-ios
# Prevent signups to WordPress.com with google accounts

Closes #3365 

This is not the best possible solution to the issue, but I think it strikes a reasonable balance at this moment.
<img src="https://user-images.githubusercontent.com/2722505/102567630-dd24d380-411c-11eb-8d2c-d24f18b0bb70.gif" width="350"/>

<img src="https://user-images.githubusercontent.com/2722505/102567620-d72ef280-411c-11eb-9767-b3ee2b0ce696.gif" width="350"/>

What I have done is block signing up to wp.com accounts with a google account. This goes in line with the spirit of some of the other changes we have implemented to the login flow on WooCommerce. 

We could (and probably should in phase 2) present a custom UI instead of the UI presented by WPAuthenticator (that is the screen with the "Try with another email" and "try with site address" buttons, but that would require reaching deep into WPAuthenticator, which seems a bit risky at this point. 

Anyway, I have passed a flag to WPAuthenticator to block account creation with a google account and that seems to work reasonably well. 

A couple of things to have in mind:
* Customizing the text in the two buttons ("Try with another email" and "Try with site address") would require reaching into WPAuthenticator deep as well. 
* I have logged #3377 to implement a proper, custom UI for this error screen, so we don't loose track of this for phase 2.

## Changes
* Pass a flag to WordPressAuthenticator to prevent signing up to a WordPress.com account with a google account. 

## How to test
* Check out the branch. Run `bundle exec pod install` just in case.
* Log out if necessary
* Tap "Continue with WordPress.com". Tap "Continue with Google"
* Log into a Google account not linked to a WordPress.com account.
* Notice the error screen. 
* On the error screen, tap "Try with another email" That should navigate to the root of the login process
* Tap "Try with the site address". That should lead to the screen to enter the Store Address.

Update release notes:
